packages/nextjs/app/checkinForm/page.tsx

```diff
@@ -1,20 +1,148 @@
 "use client";
 
-import React from "react";
+import React, { SyntheticEvent, useContext, useState } from "react";
+import { useRouter } from "next/navigation";
+import { SchemaEncoder } from "@ethereum-attestation-service/eas-sdk";
 import type { NextPage } from "next";
+import { useAccount } from "wagmi";
+import easConfig from "~~/EAS.config";
+import { EASContext } from "~~/components/EasContextProvider";
+import Mapbox from "~~/components/Mapbox";
 
 // import Link from "next/link";
 
-const CheckinFrom: NextPage = () => {
+const CheckinForm: NextPage = () => {
+  // NextJS redirect
+  const { push } = useRouter();
+
+  const { address: connectedAddress } = useAccount(); //get address from wagmi
+  const nowInSeconds = Math.floor(Date.now() / 1000);
+  const [formValues, setFormValues] = useState({
+    coordinateInputX: 0, // to be picked up by prop
+    coordinateInputY: 0, // to be picked up by prop
+    timestamp: nowInSeconds,
+    data: "Something you wanna add.",
+  });
+
+  // Use EAS SDK
+  const { eas, isReady } = useContext(EASContext);
+  // const [attestation, setAttestation] = useState<Attestation>();
+
+  // Initialize SchemaEncoder with the schema string
+  const schemaEncoder = new SchemaEncoder("string[] coordinates,address subject,uint256 timestamp,bytes32 message");
+  const encodedData = schemaEncoder.encodeData([
+    {
+      name: "coordinates",
+      value: [formValues.coordinateInputX.toString(), formValues.coordinateInputY.toString()],
+      type: "string[]",
+    },
+    { name: "subject", value: connectedAddress || "0xA332573D0520ee4653a878FA23774726811ae31A", type: "address" },
+    { name: "timestamp", value: formValues.timestamp, type: "uint256" },
+    { name: "message", value: formValues.data, type: "bytes32" },
+  ]);
+
+  const schemaUID = easConfig.SCHEMA_UID_SEPOLIA; // TODO: read according to chainId
+
+  function handleChange(event: React.ChangeEvent<HTMLInputElement>) {
+    event.preventDefault();
+    setFormValues({ ...formValues, [event.target.name]: event.target.value });
+  }
+
+  // Set attestation from EAS api
+  function handleSubmit(event: SyntheticEvent) {
+    if (!isReady) return; // notify user
+
+    event.preventDefault();
+    eas
+      .attest({
+        schema: schemaUID,
+        data: {
+          recipient: easConfig.EAS_CONTRACT_SEPOLIA, // To be read by chainId
+          expirationTime: 0n,
+          revocable: true, // Be aware that if your schema is not revocable, this MUST be false
+          data: encodedData,
+        },
+      })
+      .then(tx => {
+        return tx.wait();
+      })
+      .then(newAttestationUID => {
+        console.log("[🧪 DEBUG](newAttestationUID):", newAttestationUID);
+        push(`/attestation/uid/${newAttestationUID}`);
```

Comment:
Do we need the `uid` part of the slug here? It feels unnecessary and confusing ...

I say that because everything in EAS is represented by a UID. Removing it puts the emphasis on the `attestation` part of the slug, which makes more sense to me. (There may be a `schema` slug in the future for some reason ... idk.)